all the rubies, plus all the arches

YOU54F opened the pull request:
builds all the rubies for all the things 

- arm64/aarch64 support on linux/macos
- x86 & x86_64 support for windows
- reproducible open source release pipeline for consistent builds
- builds Ruby 2.6 through to Ruby 3.3.0
  - Only went back to 2.6.10 for unofficial HomeBrew support on Linux aarch64
- Add's a CLI installer so users can get traveling-ruby at their fingertips, if they just need a ruby runtime, without any additional packaging steps
- Tests all the things
- build musl variants of linux, both x86_64 / aarch64
- OpenSSL 3.2.0 for Ruby >= 3.1.0 / OpenSSL 1.1.1w for Ruby < 3.1.0

Fork:- https://github.com/YOU54F/traveling-ruby

background https://github.com/phusion/traveling-ruby/pull/133#issuecomment-1579665334

## Travelling ruby updates for arm64/aarch64

| OS         | Architecture | Supported |
| ------- | ------------ | --------- |
| OSX        | x86_64       | ✅         |
| OSX        | aarch64 (arm)| ✅         |
| Linux (glibc)   | x86_64       | ✅         |
| Linux (glibc)  |   aarch64 (arm)| ✅          |
| Linux (musl)  | x86_64       | ✅         |
| Linux (musl) |   aarch64 (arm)| ✅          |
| Windows | x86_64       | ✅        |
| Windows | x86       | ✅        |
| Windows| aarch64 (via x86 emulation) |  ✅        |

macos x86_64 binaries work as far back at 10.15 Catalina. 

10.14 Mojave fails on libgmp

Commit:
Bump Ruby version to 3.1.2

scrill committed:
Update Gemfile.lock

LonelyWolf committed:
added support of arm64 architecture for OSX

LonelyWolf committed:
OSX: build supported on x86_64 and arm64 platforms

YOU54F committed:
Merge branch '3.1.2' of github.com:cloudaware/traveling-ruby

YOU54F committed:
chore: Update Gemfile 20230529

YOU54F committed:
chore: Update RUBY_VERSIONS.txt to create multiple versions

YOU54F committed:
chore: update BUNDLER_VERSION to 2.4.10

YOU54F committed:
chore: Update VERSION.txt 20230529

YOU54F committed:
chore: rename script files to end with .sh

YOU54F committed:
chore: osx deployment target

YOU54F committed:
chore(osx): add 14.2 xcode path

YOU54F committed:
chore(osx): enable sqlite, postgresql, mysql, disable warning supression

YOU54F committed:
chore(osx): osx build tweaks

YOU54F committed:
chore(win): use 7zz on windows

YOU54F committed:
chore(scripts): add scripts to list, unpack and test all output versions

YOU54F committed:
chore(win): enable multi-arch builds

YOU54F committed:
chore(linux): allow cpucount to pass on darwin for x-plat builds

YOU54F committed:
feat(linux): add linux-arm64 multi-version builds

YOU54F committed:
chore(linux): build libffi and libyaml from source

YOU54F committed:
chore(linux): --disable-install-doc only, no readline on ruby 3, copy libffi into build

YOU54F committed:
chore(git): git ignores

YOU54F committed:
chore: add parser to list versions

name            pkg_date  ruby_version    os     arch   size
---             --------  --------------  --     ----   ----
traveling-ruby  20230528  2.6.10          linux  arm64  8.5M
traveling-ruby  20230528  3.0.4           linux  arm64  8.9M
traveling-ruby  20230528  3.1.0           linux  arm64  8.9M
traveling-ruby  20230528  3.1.0-preview1  linux  arm64  8.9M
traveling-ruby  20230528  3.1.1           linux  arm64  8.9M
traveling-ruby  20230528  3.1.2           linux  arm64  8.9M
traveling-ruby  20230528  3.2.2           linux  arm64  7.8M
traveling-ruby  20230528  3.3.0-preview1  linux  arm64  7.8M

YOU54F committed:
chore(ci): cirrus multi builder WIP

YOU54F committed:
cirrus packing

YOU54F committed:
chore(ci): matrix version builder and mass builder

YOU54F committed:
ci(matrix-linux-arm64): test matrix-linux-arm64

YOU54F committed:
ci(matrix-linux-arm64): use docker next image with libffi and libyaml static builds

YOU54F committed:
ci(matrix-linux-arm64): package all working ruby builds for arm64

YOU54F committed:
ci(packager-linux-arm64): use next image

YOU54F committed:
osx: working rakefile

YOU54F committed:
ci(packager-linux): package all working ruby builds for linux

YOU54F committed:
ci(packager-linux): package all working ruby builds for linux

YOU54F committed:
chore(test): remove output.txt

YOU54F committed:
chore(ruby): create RUBY_VERSIONS.txt for platforms/archs

YOU54F committed:
chore(linux): ensure rake clean deletes new arch specific gems

YOU54F committed:
feat(osx): support multi-architecture builds without using emulation

YOU54F committed:
ci(packager): windows and macos packagers

YOU54F committed:
feat(gems): update Gemfile to phusion/traveling-ruby state, exclude mysql2 for linux

YOU54F committed:
chore: reinstate linux Rakefile

YOU54F committed:
chore(linux): bundle deprecated warning - set bundle config set local system

YOU54F committed:
ci(packager-linux): test linux packager

YOU54F committed:
ci(matrix): test matrix runners

YOU54F committed:
ci(matrix): test matrix runners

YOU54F committed:
ci(matrix): split matrix runners to latest and alt

YOU54F committed:
ci(matrix): matrix runners for latest, and latest working

YOU54F committed:
ci(matrix): matrix upload task name gems

YOU54F committed:
ci(matrix): matrix upload task name gems

YOU54F committed:
ci(matrix): osx - update Rakefile to support running indiv ruby versions

YOU54F committed:
ci(matrix-macos): osx - install bundler 2.4.10 - note osx unsets user path so uses system ruby version 2.6.10

YOU54F committed:
ci(matrix-macos): osx - cirrus image has rbenv so we need to set system before install bundler

YOU54F committed:
ci(matrix-macos): osx - skip mysql2/nokigiri - add gem platforms

YOU54F committed:
chore(test): add support for testing additional pact gems

YOU54F committed:
chore(osx): macosx dep target 12.2 + cmake osx arch env var

YOU54F committed:
chore(osx): move ARCHITECTURE check to top of script, add DEPLOY_TARGET for libgmp

YOU54F committed:
chore(osx): delete 21M libruby.3.2-static.a, reducing bundle size by half

YOU54F committed:
chore(osx): start supporting building x86_64 runtime on arm64 without rosetta

YOU54F committed:
chore(osx): runtime built for x86_64 on arm64 host

YOU54F committed:
chore(osx): pass deploy target when building ruby

YOU54F committed:
chore(osx): use pre-packed/pre-published runtime

YOU54F committed:
ci(matrix-macos): osx - reduce size of bundle

YOU54F committed:
ci(matrix): test out matrix

YOU54F committed:
ci(matrix-macos-arm64): publish arm64 assets

YOU54F committed:
ci(packager-windows): publish assets

YOU54F committed:
ci(matrix-macos-arm64): publish arm64 assets

YOU54F committed:
ci(matrix-macos-arm64): publish arm64 assets

YOU54F committed:
feat(linux): aarch64 unofficial support via you54f/traveling-ruby

YOU54F committed:
chore(linux): correct CPUCOUNT

YOU54F committed:
feats(gems): perform ruby gems update on packaging

YOU54F committed:
feat(linux): create and publish multi-arch Dockerfiles

YOU54F committed:
feat(osx): add universal-darwin to bundle to support multi-arch

YOU54F committed:
feat(install): add CLI installer and GitHub Action

YOU54F committed:
chore(ci): remove system installed libs from runner

YOU54F committed:
chore(ci): remove system installed libs from runner

YOU54F committed:
chore(ci): macos add versions matrix

YOU54F committed:
chore(ci): homebrew ruby residual in path in step

YOU54F committed:
feat(linux): LinuxBrew multi-arch docker

YOU54F committed:
chore(rel): 20230605

YOU54F committed:
feat(linux): LinuxBrew multi-arch docker powered by traveling-ruby

YOU54F committed:
ci(matrix): rel-20230601

YOU54F committed:
deps(gems): update rubygems to 3.3.3

YOU54F committed:
chore(ci): update tags to main

YOU54F committed:
chore(matrix): multi-arch args osx/linux, renable sanity checks

YOU54F committed:
ci(matrix): produce -full bundle for macos/linux containing all native exts

YOU54F committed:
deps(gems): force_ruby_platform as per bundler >2.3.14

YOU54F committed:
chore: some cleanup

YOU54F committed:
chore(osx): nokogiri link to libxml2 headers, remove some files from output

YOU54F committed:
chore(linux): update linuxbrew docker file

YOU54F committed:
chore(install): setup installer to set first in GH path

YOU54F committed:
chore(ci): split out test files

YOU54F committed:
chore(osx): download libxml2 from gnome https source

YOU54F committed:
feat(linux): use shellchecked sh wrappers

YOU54F committed:
ci(matrix): rel-20230605

YOU54F committed:
chore(rdoc): dont remove rdoc generators

YOU54F committed:
ci(matrix): dont remove rdoc/provide support for installing gems at cli install time

YOU54F committed:
chore(install): make sh compliant

YOU54F committed:
chore(deps): bundler 2.4.18 plus Gemfile update - rel20230803

YOU54F commented:
Latest release has just gone out

https://github.com/YOU54F/traveling-ruby/releases/tag/rel-20230803

YOU54F commented:
@FooBarWidget This is pretty solid now, we've been running it in production in a variety of applications and not had any negative reports

YOU54F commented:
If you land here, and you are after Windows support with native gems, and Ruby 3.2.2, check out this fork https://github.com/Largo/ocran of https://github.com/larsch/ocra

dbackeus commented:
For the record we have used this fork to [release the CLI tool](https://github.com/reclaim-the-stack/k/releases/tag/5f5c911) for https://reclaim-the-stack.com on multiple platforms with 0 problems. Thanks a lot for putting in all the work @YOU54F

Would be great to get this merged to avoid reliance on forks.

YOU54F commented:
> For the record we have used this fork to [release the CLI tool](https://github.com/reclaim-the-stack/k/releases/tag/5f5c911) for https://reclaim-the-stack.com on multiple platforms with 0 problems. Thanks a lot for putting in all the work @YOU54F
> 
> Would be great to get this merged to avoid reliance on forks.

Awesome, super pleased it is working out for you! <3

Happy to make any changes to the PR to help with assisting getting it merged.

I think we will want to upgrade openssl to v3 as v1.1.x is now EOL

https://www.openssl.org/blog/blog/2023/09/11/eol-111/

YOU54F committed:
deps: ruby 3.3.0 / bundler 2.5.3

YOU54F committed:
deps: macos latest deps

YOU54F committed:
linux: use updated holy-build-box openssl3.2

YOU54F committed:
ci: add gh tasks

YOU54F committed:
linux: update openssl/extconf.rb for ruby 3.3 and below

YOU54F committed:
ci: ruby 3.3.0 in gh workflows

YOU54F committed:
ci: ruby 3.3.0 in gh workflows

YOU54F committed:
ci: only ruby 3.3.0 in gh workflows

YOU54F committed:
ci: arm64 xcode version m1

YOU54F committed:
chore: remove versions of ruby incompatible with openssl3

YOU54F committed:
ci: macos use new runtime

YOU54F committed:
osx: link ncurses libs properly

YOU54F committed:
linux/macos: enable mysql2

YOU54F committed:
osx: mysql2 connecter fix with SO link

YOU54F committed:
win: dont skip 3.3.0 from cirrus

YOU54F committed:
ci: github action wip / skip arm64 linux

YOU54F committed:
osx: download gmp from diff lcn as blocked from GHA

YOU54F committed:
ci: increase cirrus execution time for x86_64 task

YOU54F committed:
linux: x86_64 set /hbb_shlib/lib64 LDFLAGS

YOU54F committed:
osx: use prepacked runtime

YOU54F committed:
osx: prepacked runtime arch env var incorrect

YOU54F committed:
osx: rmeove setup-xcode from arm64 gha

YOU54F committed:
linux: missing / on pkg config path

YOU54F committed:
ci: arm64 gha support / cirrus download runtime paths

YOU54F committed:
ci: incorrect path in osx script

YOU54F committed:
ci: arm64 test path

YOU54F committed:
ci: osx dl runtime path fix

YOU54F committed:
ci: osx dl runtime path fix

YOU54F committed:
feat: add support for openssl 1.1.1 builds to support ruby 3.0 and lower

YOU54F committed:
osx(ci): upload gha osx runtime

YOU54F committed:
ci: use uploaded osx gha runtime to speed up runs

YOU54F committed:
ci: gh build ruby 3.1/3.2/3.3

YOU54F committed:
ci: gh build ruby windows all versions

YOU54F committed:
macos: create osx openssl1.1.1 based runtime

YOU54F committed:
macos: create osx openssl1.1.1 based runtime

YOU54F committed:
macos: use osx openssl1.1.1 pre baked runtime

YOU54F committed:
macos: use osx openssl1.1.1 pre baked runtime

YOU54F committed:
ci(matrix): test multiple ruby versions inc openssl 1.1.1

YOU54F committed:
ci(matrix): test multiple ruby versions inc openssl 1.1.1

YOU54F committed:
ci(matrix): remove all broken matrix versions

YOU54F committed:
rel: 20240205 - Ruby 3.3.0 + OpenSSL 3 for Ruby 3.1.x +

YOU54F committed:
feat: create musl variants with alpine holy build box

YOU54F committed:
update release date

YOU54F committed:
alpine: update libcheck for libc.musl-x86_64

YOU54F committed:
ci:dont build image

YOU54F committed:
ci: update cirrus to build alpine

YOU54F committed:
ci(alpine): test alpine matrix on linux

YOU54F committed:
Merge pull request #2 from YOU54F/alpine

Alpine

YOU54F committed:
chore: rm .ruby-version and revert bundler to 2.5.3

YOU54F commented:
New release is out

## Highlights

- New rubies
- New gems (mysql2) plus Gem updates
- Bundler update
- Runtime dep updates
- New platforms
  - New Alpine based holy-build box and musl based linux packages
- New GH Workflows
  - Updated GH action workflows, including macos-14 
 
## New Rubies

- Ruby 3.3.0 Release

## Bundler Update

- Bundler updated to 2.5.3 for Ruby versions > 3
- Bundled updated to 2.4.22 for Ruby versions < 3 

## New Gems plus Gem updates

- mysql2 gem now prebuilt for linux/macos/alpine

```
    RedCloth (4.3.3)
    bcrypt (3.1.20)
    charlock_holmes (0.7.7)
    curses (1.4.4)
    daemons (1.4.1)
    escape_utils (1.3.0)
    eventmachine (1.2.7)
    fast-stemmer (1.0.2)
    ffi (1.16.3)
    hitimes (2.0.0)
    json (2.7.1)
    kgio (2.11.4)
    mini_portile2 (2.8.5)
    mysql2 (0.5.6)
    nio4r (2.7.0)
    nokogiri (1.16.2)
      mini_portile2 (~> 2.8.2)
      racc (~> 1.4)
    pg (1.5.4)
    posix-spawn (0.3.15)
    puma (6.4.2)
      nio4r (~> 2.0)
    racc (1.7.3)
    rack (2.2.8)
    raindrops (0.20.1)
    redcarpet (3.6.0)
    rexml (3.2.6)
    rugged (1.7.2)
    sqlite3 (1.7.2)
      mini_portile2 (~> 2.8.0)
    thin (1.8.2)
      daemons (~> 1.0, >= 1.0.9)
      eventmachine (~> 1.0, >= 1.0.4)
      rack (>= 1, < 3)
    unf_ext (0.0.9.1)
    unicorn (6.1.0)
      kgio (~> 2.6)
      raindrops (~> 0.7)
    yajl-ruby (1.4.3)
```

## OpenSSL

Ruby versions lower than 3.1.0 are built with OpenSSL 1.1.1w
Ruby versions 3.1.0 or later are build with OpenSSL 3.2.0

## Dependency Updates

### Linux Runtime updates

CCACHE_VERSION=4.9
CMAKE_VERSION=3.28.1
CMAKE_MAJOR_VERSION=3.28
GCC_LIBSTDCXX_VERSION=9.3.0
ZLIB_VERSION=1.3
OPENSSL_VERSION=3.2.0
CURL_VERSION=8.5.0
GIT_VERSION=2.43.0
SQLITE_VERSION=3450000
SQLITE_YEAR=2024
MYSQL_LIB_VERSION=6.1.9
POSTGRESQL_VERSION=15.5
ICU_RELEASE_VERSION=74-1
ICU_FILE_VERSION=74_1
LIBSSH2_VERSION=1.11.0
LIBFFI_VERSION=3.4.4
LIBYAML_VERSION=0.2.5

### MacOS Runtime updates

CCACHE_VERSION=4.9
CMAKE_VERSION=3.28.1
PKG_CONFIG_VERSION=0.29.2
AUTOCONF_VERSION=2.71
AUTOMAKE_VERSION=1.16.5
LIBTOOL_VERSION=2.4.7
OPENSSL_VERSION=3.2.0
NCURSES_VERSION=6.4
LIBEDIT_VERSION=20230828-3.1
LIBEDIT_DIR_VERSION=20230828-3.1
GMP_VERSION=6.3.0
GMP_DIR_VERSION=6.3.0
LIBFFI_VERSION=3.4.4
LIBYAML_VERSION=0.2.5
SQLITE3_VERSION=3450000
SQLITE3_VERSION_YEAR=2024
XZ_VERSION=5.4.5
MYSQL_LIB_VERSION=6.1.9
POSTGRESQL_VERSION=15.5
ICU_RELEASE_VERSION=74-1
ICU_FILE_VERSION=74_1
LIBSSH2_VERSION=1.11.0
LIBXML2_VERSION=2.9.14
LIBXSLT_VERSION=1.1.34

## New Platforms 

musl based traveling-ruby packages

- Platforms
  - aarch64-linux-musl
  - x86_64-linux-musl
- Uses alpine flavoured holy build box
  - https://github.com/YOU54F/holy-build-box/tree/alpine
  - Images published to DockerHub with `-alpine` suffix
    - https://hub.docker.com/r/you54f/holy-build-box/tags?page=1&name=alpine
  - Builds against Alpine 3.15. Tested against 3.15 - 3.19
  - Currently only built for OpenSSL 3.2.0 so Ruby 3.1.x +
- Alpine Traveling Ruby builder images published to docker, with `-alpine` suffix
  - https://hub.docker.com/r/you54f/traveling-ruby-builder-arm64/tags?page=1&name=alpine
  - https://hub.docker.com/r/you54f/traveling-ruby-builder-x86_64/tags?page=1&name=alpine

YOU54F commented:
So I took a diversion from Ruby for a bit, to port one of our Ruby apps to Rust.

I thought it would be nice to have a fall back in the cli for each command, to allow it to call the existing ruby cli app, by use of a flag or env var.

With the help of some cross-compilers, I was able to package the rust app, for more supported platform/arch combos than ruby.

That left me with the behaviour of having to tell users who are wanting to fallback to the ruby impl, but don't have a traveling-ruby runtime that they are 💩 out of luck, and stuck with my poorly written rust code. 🤮 

What to do 🤔 

💡 cross-compilation adventure with qemu, and a switch to ubuntu/alpine over centos 7, as they had multi-platform images available where centos 7 we were stuck only working builds for amd64/arm64.

## Updates

New additions

- Linux - i386 / s390x / ppc64le / riscv64
- Musl variants of all Linux architectures
- Windows arm64 (Ruby 3.1.4 only)
- New gems (bigdecimal / psych / stringio)
- support for building snapshot from branches or master from ruby releases for (macos/linux) or head from rubyinstaller2 releases (windows)

🚧- Your mileage may vary with new platforms
⚠️ - openssl 1.1 based builds have been dropped, so nothing earlier than 3.1.4 for linux/macos

Check out the pre-release [here](https://github.com/YOU54F/traveling-ruby/releases/tag/rel-20240412)

### Linux

| Platform                 | Architecture  | Musl | Glibc
|--------------------------|---------------|------|-------
| ARMv8 64-bit (`arm64v8`) | ARMv8         |  ✅  | ✅
| Linux x86-64 (`amd64`)   | x86-64        |  ✅  | ✅
| x86/i686 (`i386`)        | x86/i686      |  ✅  | ✅
| IBM z Systems (`s390x`)  | z Systems     |  ✅  | ✅
| IBM POWER8 (`ppc64le`)   | POWER8        |  ✅  | ✅
| RISC-V 64-bit (`riscv64`)| RISC-V        |  ✅  | ✅ 

- Alpine binaries built against [3.15](https://hub.docker.com/layers/library/alpine/3.15/images/sha256-26284c09912acfc5497b462c5da8a2cd14e01b4f3ffa876596f5289dd8eab7f2)
  - [riscv64](https://hub.docker.com/layers/riscv64/alpine/20210804/images/sha256-2387fa34569bac5a2745bbe536105834be0c5a72be9d0fa1c793403672c220a2?context=explore) built against 3.15 alpha 
- Ubuntu binaries built against [14.04](https://hub.docker.com/layers/library/ubuntu/14.04/images/sha256-534255069ef3adf1c7a555fd9f614845ab66c61ef9846e438d86e12ae8c89b88) (glibc 2.19)
  - ppc64le / s390x binaries against [16.04](https://hub.docker.com/layers/library/ubuntu/16.04/images/sha256-a3785f78ab8547ae2710c89e627783cfa7ee7824d3468cae6835c9f4eae23ff7)
  - riscv64 binaries against [20.04](https://hub.docker.com/layers/riscv64/ubuntu/20.04/images/sha256-c0e96f609e5f128bdb6d24d21d3bf0a9af6f17234976fa574dad7146a0f97fb2?context=explore)

## MacOS

| Platform                 | Architecture  | Supported
|--------------------------|---------------|------
| MacOS x86-64 (`darwin-x86_64`) | x86-64 |  ✅  
| MacOS arm64 (`darwin-arm64`)   | arm64     |  ✅

- macos x86_64 binaries - 10.15 Catalina onwards
- macos arm64 binaries - 11.0 Big Sur onwards

## Windows

| Platform                 | Architecture  | Supported
|--------------------------|---------------|------
| Windows x86-64 (`windows-x86_64`) | x86-64 |  ✅  
| Windows x86 (`windows-x86`)   | x86     |  ✅
| Windows arm64 (`windows-arm64`)   | arm64     |  🚧

- windows-arm64, ruby 3.1.4 only
- 🚧 Native extensions not currently supported
  - Use ocran or aibika (forks of ocran) to build native extensions
- Docker Support
  - Nanoserver images, will work if libgmp from package is copied to C:\Windows\System32
- Wine support
  - x86_64 package fails on darwin-arm64 with unexpected ucrtbase.dll error
    - Workaround, use x86 package on darwin-arm64
- Windows VM support
  - x86_64 package fails when emulated in vm's on darwin-arm64 with unexpected ucrtbase.dll error
    - Workaround, use x86 package on darwin-arm64
    - Workaround, use arm64 package on darwin-arm64

YOU54F commented:
Also I've been working on a traveling-ruby app builder built in rust, for end users, that will package up their app for any or all of the supported TR platforms. 

Rust supports embedding files (so an archive) or directory (with rust-embed), and I've managed to eliminate the need for end users app wrappers, by directly invoking the ruby binary with the correct TR environment variables setup. 

This means you can distribute your ruby app as a single file which self-extracts its contents and then executes.

It's similar in vein for ocra / ruby_packer which provide a single entry point, however each of those unpack to a temporary directory, which affects speeds (it takes a short while to unpack it every time, and that seems expensive for our use case, cli invocations of which there are highly likely to multiple in quick succession)

In order to support native extensions for windows in my end users app, I cheated a little bit, I used ocran, exploded the temp directory and provided a new wrapper script.

Not sure if it will be of use to others, but it's been scratching my itch of getting to learn more rust and made it easy for me to test out the new TR combos

utoppo commented:
> Also I've been working on a traveling-ruby app builder built in rust, for end users, that will package up their app for any or all of the supported TR platforms.
> 
> Rust supports embedding files (so an archive) or directory (with rust-embed), and I've managed to eliminate the need for end users app wrappers, by directly invoking the ruby binary with the correct TR environment variables setup.
> 
> This means you can distribute your ruby app as a single file which self-extracts its contents and then executes.
> 
> It's similar in vein for ocra / ruby_packer which provide a single entry point, however each of those unpack to a temporary directory, which affects speeds (it takes a short while to unpack it every time, and that seems expensive for our use case, cli invocations of which there are highly likely to multiple in quick succession)
> 
> In order to support native extensions for windows in my end users app, I cheated a little bit, I used ocran, exploded the temp directory and provided a new wrapper script.
> 
> Not sure if it will be of use to others, but it's been scratching my itch of getting to learn more rust and made it easy for me to test out the new TR combos

That really sounds great! Thank you so much for all the effort!

YOU54F committed:
chore(deps): latest round of rubies due to cves

updates rubygems to 3.5.9
updates bundler to 2.5.9

note need to drop 3.1.7 openssl fails to build in linux, (3.1.4 okay)

YOU54F committed:
!chore: drop ruby 3.0.x and lower, remove openssl 1.1

YOU54F committed:
deps: add bigdecimal/psych/stringio

YOU54F committed:
osx: fixup build errors

YOU54F committed:
chore(deps): ruby 3.3.2 / 3.1.6 / rubygems 3.5.11 / bundler 2.5.11

plus gem updates

ffi 1.17.0 (was 1.16.3)
racc 1.8.0 (was 1.7.3)
mini_portile2 2.8.7 (was 2.8.6)
hitimes 3.0.0 (was 2.0.0)
nio4r 2.7.3 (was 2.7.1)
curses 1.4.6 (was 1.4.5)
nokogiri 1.16.5 (was 1.16.4)
sqlite3 2.0.2 (was 2.0.1)
rexml 3.2.9 (was 3.2.6)

YOU54F committed:
chore(deps): ruby 3.3.3

## fixes
- mini_portile2 pin to 2.8.6 due to errors in 2.8.7 on macos

## gem updates

rexml 3.3.0 (was 3.2.9)
stringio 3.1.1 (was 3.1.0)
nokogiri 1.16.6 (was 1.16.5)

YOU54F committed:
chore(ci): use setup-ruby@v1.180.0 for 3.3.3 on windows [skip ci]

YOU54F committed:
deps: ruby 3.3.4

YOU54F committed:
!deps(gems): gem updates charlock_holmes/rexml

charlock_holmes (0.7.9)
rexml (3.3.2)

drop unmaintained posix-spawn

YOU54F committed:
chore(ci): update ruby setup actions for ruby 3.3.4

YOU54F committed:
chore(test): remove posix-spawn from test matrix

YOU54F committed:
macos: add xcode 15.4 sdk path

YOU54F committed:
chore: update gems + ruby 3.3.5 / 3.2.5 + bundler 2.5.18

YOU54F committed:
chore: drop windows to 3.3.4 / update rubygems version to 3.5.18

YOU54F committed:
chore: remove platform specific ffi gems from lockfile

YOU54F committed:
deps(win): bump to ruby 3.3.5

bradgessler commented:
I think I'm going to use this for https://terminalwire.com/, which uses the async Ruby libraries, including async-websockets. 😅

YOU54F committed:
chore(deps): ruby 3.2.6 / 3.3.6

YOU54F committed:
chore(deps): updated gems
-    curses (1.4.6)
+    curses (1.4.7)
-    json (2.7.2)
+    json (2.8.2)
-    nio4r (2.7.3)
+    nio4r (2.7.4)
-    pg (1.5.7)
-    psych (5.1.2)
+    pg (1.5.9)
+    psych (5.2.0)
-    puma (6.4.2)
+    puma (6.4.3)
-    rack (2.2.9)
+    rack (2.2.10)
-    rexml (3.3.7)
+    rexml (3.3.9)
-    sqlite3 (2.0.4)
+    sqlite3 (2.3.0)
-    stringio (3.1.1)
+    stringio (3.1.2)

YOU54F committed:
chore: rel-20240904

YOU54F committed:
chore(osx): update codepaths for 15.4 & 16.0 xcode (gha / cirrus)

YOU54F committed:
chore(osx): build osx in gha on macos-13 as macos-12 will brownout

YOU54F committed:
chore(osx): downgrade x86_64 macos-13 xcode to 15.2

YOU54F committed:
chore(osx): use xcode 16 for cirrus

YOU54F committed:
chore(deps): update gems

Using bigdecimal 3.1.9 (was 3.1.8)
Using rexml 3.4.0 (was 3.3.9)
Using json 2.9.1 (was 2.8.2)
Using psych 5.2.2 (was 5.2.0)
Using puma 6.5.0 (was 6.4.3)

YOU54F committed:
chore(rel): update to 20241228

YOU54F committed:
chore(ruby): add 3.4.1

YOU54F committed:
chore(ci): update to setup-ruby@v1.206.0

YOU54F committed:
chore(osx): use xcode 16.1 for cirrus

YOU54F committed:
chore(deps): remove ffi platform specific gems from lockfile

YOU54F committed:
chore(deps): all gems + latest rubies
3.4.5 / 3.3.7 / 3.2.7

YOU54F committed:
ci: update to v4 action tags

YOU54F committed:
ci: update to latest setup-ruby

YOU54F committed:
ci: update to latest setup-ruby

YOU54F committed:
ci: downgrade to 3.4.4 as rubyinstaller2 doesnt have .5 yet

YOU54F committed:
chore(osx): update libxslt dl location

YOU54F committed:
chore(osx): use 15.2 xcode on gh

YOU54F committed:
chore(deps): pin psych to 5.2.0

fails to strip on macos 3.4.x ruby versions

YOU54F committed:
chore(osx): dont strip dsym bundles on 3.4.x

YOU54F committed:
chore(osx): add xcode 16.4 for osx

YOU54F committed:
chore(win): set exe bit on gem executable

YOU54F committed:
chore(deps): 3.2.8 / 3.3.8 / 3.4.5

YOU54F committed:
chore(win): build 3.4.4-2

3.4.5 not yet avail

YOU54F committed:
chore(deps): add ruby 3.4.5 / 3.3.8 / 3.2.8

YOU54F committed:
chore(deps): update gemfile

YOU54F committed:
chore(ci): use setup-ruby main for latest rubies

YOU54F committed:
chore(deps): remove native ffi exts from shared gemfile

YOU54F committed:
chore(ci): use setup-ruby master for latest rubies

YOU54F committed:
feat(win): arm64 support 3.4.x +

YOU54F committed:
ci(packager-windows): build for arm64

YOU54F committed:
ci(packager-windows): build for arm64 - fix cirrus script

YOU54F committed:
ci(packager-windows): build for arm64 - fix cirrus script

YOU54F commented:
@FooBarWidget - I guess you aren't interested in the PR!

Might as well archive it now fella!

From my perspective, I have am pretty much wrapping up a rewrite of the tooling which depended on this package, so my inclination to maintain it will wane, as I am cleaning up my digital garden.

It is pretty automated now, and we have much better cross platform support in CI in github actions which pretty much negates the need for cirrus-ci

FooBarWidget commented:
Hello @YOU54F, I want to let you know that I haven't forgotten your request. I have done my upmost best in the past few months to allocate time, but it just doesn't work out. Between needing to take care of spouse, children, parents, and health issues, I just can't do it. My health is not cooperating and I can't push myself that hard anymore. There are too many obligations and not enough time and health points.

I can only advise not to rely too much on me. If you feel strongly about something, go ahead and take matters into your own hands. Fork or start your own project if you want. I'm afraid that for now I have reached the limits of what I can do.

YOU54F commented:
Thanks Hongli, for everything you've done for the project and the ruby community. I can empathise fully on the health situation with family. 

I hope you can maximise the time to spend with them all.

Well if it helps to lift a weight off your mind, you can forget about this pull request, and know that the traveling-ruby torch is being carried on, in an automated fashion, and will hopefully continue on for as long as the operating systems we run on, co-operate!

I might make an org for this, and some other ruby packaging tools, I've quite a bit of work across a fair few packagers and it would be a shame to lose it when other intrepid open sorcerers might want to turn their hand to any one of them.

We at the pact-foundation, have been eternally grateful for traveling-ruby, it enabled pact-ruby to be used in many different languages, spawned life long friendships, careers, relationships, marriages, jobs in open-source.

So thank you from all of the Pact team. Live long and prosper dude ✌🏾

dbackeus commented:
@YOU54F I've had a lot of value of this PR (we use it for the packaging of our CLI tool for the https://reclaim-the-stack.com project). I didn't fully follow what your intent is now after closing. Do you have an alternative packager in use within Pact which you're considering releasing?

In any case. Thanks for your work ❤️

Also thank you @FooBarWidget for initiating this project. Best wishes for your health and family 🙏

YOU54F commented:
The new home is https://github.com/trubygems/traveling-ruby and the fork has been pulled out of the network for visibility.

The rewritten Pact tooling, is in Rust and packaged as a single binary. It doesn't rely on the Pact ruby tooling packaged with Ruby, but does offer downloading those traveling-ruby created packages for legacy compatibility.

For those legacy packages, I still rely on the binaries generated as part of this pull request so have done some work to make the CI process as slick as possible for the future, so it can be easily be maintained for a long time.

Latest release went out over the weekend - https://github.com/trubygems/traveling-ruby/releases/tag/rel-20251107

I do have plans for improvements to the project, which I will announce soon, but you can rest safe that if you are relying on the project, that existing releases will remain, and you should be able to fork and generate your own binaries easily in github for safe keeping! ( I would consider this a smart idea, if used in production pipelines )
